File: ProblemSolving/450DSA/Python/src/bitmanipulation/PowerOfTwo.py
"""
@author Anirudh Sharma

Given a non-negative integer N. The task is to check if N is a power of 2.
More formally, check if N can be expressed as 2^x for some x.
"""


def isPowerOfTwo(n):
    if n <= 0:
        return False
    return (n & (n - 1)) == 0


if __name__ == "__main__":
    print(isPowerOfTwo(2097152))
    print(isPowerOfTwo(8336))
    print(isPowerOfTwo(1))
    print(isPowerOfTwo(0))

File: ProblemSolving/450DSA/JavaScript/src/bitmanipulation/PowerOfTwo.js
/**
 * @author Anirudh Sharma
 * 
 * Given a non-negative integer N. The task is to check if N is a power of 2.
 * More formally, check if N can be expressed as 2^x for some x.
 */
const isPowerOfTwo = (n) => {
    if (n <= 0) {
        return false;
    }
    return (n & (n - 1)) == 0;
};

const main = () => {
    console.log(isPowerOfTwo(2097152));
    console.log(isPowerOfTwo(8336));
    console.log(isPowerOfTwo(1));
    console.log(isPowerOfTwo(0));
};

main();